### PR TITLE
XD-2834: generate docs as part of "standard" build

### PR DESCRIPTION
**Pull request by ericbottard**
* Refactored docs related tasks to reside where they logically belong
* Make them part of the standard build (a prereq of `./gradlew build`)
* Change ShellReference doc generation to force iteration order (should remove false edits that appear depending on which JDK is running the task)

### CODE REVIEW

File: documentation-toolchain/src/test/resources/hadoop.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+#Exists do that ShellReferenceDoc does not blow up when loading XD Shell's META-INF/spring/spring-shell-plugin.xml
```
Comment (ghillert) [inline documentation-toolchain/src/test/resources/hadoop.properties]: You mean `Exists so that` ?

File: build.gradle [hunk 2]
```diff
@@ -1375,3 +1367,7 @@ idea.project.ipr {
 		provider.node.component.find { it.@name == 'VcsDirectoryMappings' }.mapping.@vcs = 'Git'
 	}
 }
+
+task build(dependsOn: ':documentation-toolchain:checkDocsLinks') {
+	
+}
```
Comment (ghillert) [inline build.gradle]: missing return

File: gradle/build-docs.gradle [hunk 3]
```diff
@@ -14,37 +14,52 @@ buildscript {
...
+	task moduleOptionsReferenceDoc(type: JavaExec) {
+		classpath = project(':spring-xd-shell').sourceSets.test.runtimeClasspath
+		main = 'org.springframework.xd.shell.util.ModuleOptionsReferenceDoc'
+		workingDir = rootProject.file('.')
+		args = [
+				"$docsDir/Sources.asciidoc",
+				"$docsDir/Processors.asciidoc",
+				"$docsDir/Sinks.asciidoc",
+				"$docsDir/Batch-Jobs.asciidoc",
+				"$docsDir/Analytics.asciidoc",
```
Comment (ghillert) [inline gradle/build-docs.gradle]: This needs to be hard-coded? Might be okay, just double-checking :-)

File: documentation-toolchain/src/test/resources/hadoop.properties (hunk 1 above)
Comment (ericbottard) [inline documentation-toolchain/src/test/resources/hadoop.properties]: yes :)

File: gradle/build-docs.gradle (hunk 3 above)
Comment (ericbottard) [inline gradle/build-docs.gradle]: Yes, this has always been like so. Avoids scanning all asciidoc files

**Comment by ghillert** [comment]
@ericbottard Let me know if you plan on committing those 2 trivial changes above or if I shall address those during the merge.

**Comment by ghillert** [comment]
Also, maybe rebase the code.

**Comment by ericbottard** [comment]
In the process of doing that.

Realized that I forgot to move ModuleOptionsReferenceDoc out of shell.
Will ping you once this PR is updated correctly

**Comment by ericbottard** [comment]
Addressed all issues

**Comment by ghillert** [comment]
LGTM - merging

**Pull request closed without merging**